src/server/config.rs

```diff
@@ -157,7 +164,15 @@ impl Config {
                 self.advertise_addr
             ));
         }
-
+        if !self.status_addr.is_empty() {
+            box_try!(config::check_addr(&self.status_addr));
+        }
+        if self.status_addr == self.advertise_addr {
```

Comment:
Is this check necessary?

Reply:
Since it's not legal, I think it is better to be checked here.

Reply:
em, please add a test for it.